File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/GroupChat.java
package dev.skyherobrine.service.models.mongodb;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "group_chats")
@Getter @Setter @NoArgsConstructor @RequiredArgsConstructor
public class GroupChat {
    @MongoId @Field(targetType = FieldType.INT64) @NonNull
    private Long id;

    @Field(targetType = FieldType.STRING) @NonNull
    private String senderId;

    @Field(targetType = FieldType.STRING) @NonNull
    private String groupId;

    @Field(targetType = FieldType.STRING) @NonNull
    private String message;

    @Field(name = "is_recall", targetType = FieldType.BOOLEAN) @NonNull
    private Boolean isRecall;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(name = "created_at", targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(name = "updated_at", targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime updatedAt;

}

File: healthcare_app_server/chat_service/models/mongodb/group-chat.js
class GroupChat {
    constructor(id, senderId, groupId, message) {
        this._created_at = this._updated_at = new Date();
        this._is_recall = false;
        this._id = id;
        this._senderId = senderId;
        this._groupId = groupId;
        this._message = message;
    }

    get id() {
        return this._id;
    }

    get senderId() {
        return this._senderId;
    }

    get groupId() {
        return this._groupId;
    }

    get message() {
        return this._message;
    }

    get created_at() {
        return this._created_at;
    }

    get updated_at() {
        return this._updated_at;
    }

    get is_recall() {
        return this._is_recall;
    }
}